source/android/adaptivecards/src/main/java/io/adaptivecards/renderer/BackgroundImageLoaderAsync.java

```diff
@@ -70,13 +74,76 @@ public void draw(Canvas canvas)
                 case RepeatHorizontally:
                     tileHorizontally(canvas);
                     break;
-                case Stretch:
+                case Cover:
                 default:
-                    super.draw(canvas);
+                    resizeBitmapForCover(canvas);
                     break;
             }
         }
 
+        /**
+         *
+         * @param containerCanvas
+         * @param originalSizeBitmap
+         * @return
+         */
+        private double getScaleFactorForCover(Canvas containerCanvas, Bitmap originalSizeBitmap)
+        {
+            double xScaleFactor = (double)containerCanvas.getWidth() / (double)originalSizeBitmap.getWidth();
+            double yScaleFactor = (double)containerCanvas.getHeight() / (double)originalSizeBitmap.getHeight();
+
+            return Math.max(xScaleFactor, yScaleFactor);
+        }
+
+        // TODO: Optimize the rendering as drawBitmap source rectangle can scale images automatically
```

Comment:
>// TODO: Opti [](start = 8, length = 13)

if this is still true, we should file an issue in github for it, then refer to it here. otherwise, remove the comment :)